src/pages/index.js

```diff
@@ -74,43 +150,136 @@ previewModalCloseBtn.addEventListener("click", () => {
   closeModal(previewModal);
 });
 
-function getCardElement(data) {
-  const cardElement = cardTemplate.content
-    .querySelector(".card")
-    .cloneNode(true);
-  const cardNameElement = cardElement.querySelector(".card__title");
-  const cardImageElement = cardElement.querySelector(".card__image");
-  const cardLikedBtn = cardElement.querySelector(".card__like-btn");
-  const deleteButton = cardElement.querySelector(".card__delete-btn");
+let selectedCard;
+let selectedCardId;
+const deleteForm = document.querySelector("#delete-form");
 
-  cardImageElement.src = data.link;
-  cardImageElement.alt = data.name;
-  cardNameElement.textContent = data.name;
+function getCardElement(cardData) {
+  const cardTemplate = document.querySelector("#card-template").content;
+  const cardElement = cardTemplate.querySelector(".card").cloneNode(true);
 
-  cardLikedBtn.addEventListener("click", () => {
-    cardLikedBtn.classList.toggle("card__like-btn_liked");
-  });
+  // Add a unique ID to the card
+  cardElement.id = `card-${cardData._id}`;
+
+  const cardImage = cardElement.querySelector(".card__image");
+  const cardTitle = cardElement.querySelector(".card__title");
+  const cardLikeButton = cardElement.querySelector(".card__like-btn");
+  const deleteButton = cardElement.querySelector(".card__delete-btn");
+
+  cardImage.src = cardData.link;
+  cardImage.alt = cardData.name;
+  cardTitle.textContent = cardData.name;
 
   deleteButton.addEventListener("click", () => {
-    cardElement.remove();
+    handleDeleteCard(cardElement, cardData);
   });
 
-  cardImageElement.addEventListener("click", () => {
```

Comment:
* The **preview** modal doesn't open when I click on the cards https://snipboard.io/VQTPwR.jpg  . This is the image modal (preview modal) https://snipboard.io/fKxBGp.jpg


Please, add a click handler for `cardImage`, to be able to show the **preview modal**